qrexec/policy/parser_compat.py

```diff
@@ -0,0 +1,96 @@
+# The Qubes OS Project, http://www.qubes-os.org
+#
+# Copyright (C) 2019  Wojtek Porczyk <woju@invisiblethingslab.com>
+#
+# This library is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License as published by the Free Software Foundation; either
+# version 2.1 of the License, or (at your option) any later version.
+#
+# This library is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this library; if not, see <https://www.gnu.org/licenses/>.
+
+import collections
+import functools
+import logging
+import pathlib
+
+from .. import POLICYPATH_OLD
+from ..exc import PolicySyntaxError
+from . import parser
+
+@functools.total_ordering
+class _NoArgumentLastKey:
+    def __init__(self, arg):
+        self.arg = arg
+    def __eq__(self, other):
+        return self.arg == other.arg
+    def __lt__(self, other):
+        return self.arg == '*' or self.arg < other.arg
```

Comment:
This looks wrong. Files without explicit argument should be sorted after those with an argument.
```suggestion
        return other.arg == '*' or self.arg < other.arg
```

Reply:
No, that would also be wrong for `self.arg == other.arg == '*'`. I'll rewrite it as `__ge__`. In fact I may have been thinking about it.